src/ext/muda/gui/puregl/mugui_puregl.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+/**
+ * @file: ext/muda/gui/puregl/mugui_puregl.h
+ * @author: sailing-innocent
+ * @create: 2023-01-09
+ * @desp: this is an pure opengl gui interface
+*/
+
+#include "../glad/gl.h"
+#include "GLFW/glfw3.h"
+#include "../mugui_base.h"
+#include <string>
+#include "../utils/gl_shader.h"
+
+namespace muda
+{
+class MuGuiPureGL : public MuGuiBase
+{
+  public:
+    MuGuiPureGL() = default;
+    ~MuGuiPureGL();
+    MuGuiPureGL(IMuGuiMode mode,
+                std::string _vert_shader_path = "../../../../test/data/gui_assets/basic_shaders/shader.vert",
```

Comment:
```suggestion
                std::string _vert_shader_path = "",
```
Relative path as default argument is dangerous here, cuz users may launch the app anywhere.
In this case, hard code shader may be better.
```cpp
std::string default_vert_shader = R"(#version 450
layout (location=0) in vec4 aPos;
layout (location=1) in vec4 aColor;
out vec4 pColor;
void main()
{
    gl_Position = aPos;
    pColor = aColor;
})";
```